src/main/java/org/jenkinsci/plugins/workflow/graph/FlowNode.java

```diff
@@ -249,17 +250,19 @@ public String getUrl() throws IOException {
      */
     protected synchronized void setActions(List<Action> actions) {
             this.actions = new CopyOnWriteArrayList<Action>(actions);
-//            this._wrapper = new ListWrapper(this);
     }
 
     /**
-     * Return the first persistent action on the FlowNode, without consulting TransientActionFactories
-     * Used here because it is much faster to invoke a less-conditional, monomorphic method (50% faster)
+     * Return the first nontransient {@link Action} on the FlowNode, without consulting {@link jenkins.model.TransientActionFactory}s
+     * <p> This is not restricted to just Actions implementing {@link PersistentAction} but usually they should.
+     * Used here because it is much faster than base {@link #getAction(Class)} method.
      * @param type Class of action
      * @param <T>  Action type
-     * @return First persistent action or null if not found
+     * @return First nontransient action or null if not found.
      */
-    public <T extends Action> T getPersistentAction(@Nonnull Class<T> type) {
+    @CheckForNull
+    @Restricted(NoExternalUse.class)  // Limit use to workflow-api packages until we have a case where we need the performance badly.
+    public final <T extends Action> T getPersistentAction(@Nonnull Class<T> type) {
```

Comment:
Advisable to make this `<T extends PersistentAction>`; mandatory if removing the restriction.